lib/Hakyll/Core/Compiler/Internal.hs

```diff
@@ -166,30 +182,29 @@ instance MonadMetadata Compiler where
 --------------------------------------------------------------------------------
 instance MonadError [String] Compiler where
   throwError = compilerThrow
-  catchError = (. matchErr) . compilerCatch
-    where
-      matchErr f Logger.Error es = f es
-      matchErr f _            _  = f []
```

Comment:
I don't remember why I did pass no messages to the the handler here - to make it distinguishable?
Now `catchError` will also catch `NoCompilationResult`s, not just `CompilationFailure`s, which seems reasonable - they are errors, at least outside of template applications.
However, ``c `catchError` throwError`` will change the error reason type, which I am not so happy about. Maybe we should export a helper function that allows altering (adding to) the messages without changing the type. [`Bifunctor`](https://hackage.haskell.org/package/base-4.10.1.0/docs/Data-Bifunctor.html) comes to my mind, but our messages are string lists not arbitrary types.